In [1]:
import pandas as pd

In [2]:
import numpy as np

In [3]:
data=pd.read_excel('trees.xlsm')

In [4]:
data

Girth  Height  Volume
0     8.3      70    10.3
1     8.6      65    10.3
2     8.8      63    10.2
3    10.5      72    16.4
4    10.7      81    18.8
5    10.8      83    19.7
6    11.0      66    15.6
7    11.0      75    18.2
8    11.1      80    22.6
9    11.2      75    19.9
10   11.3      79    24.2
11   11.4      76    21.0
12   11.4      76    21.4
13   11.7      69    21.3
14   12.0      75    19.1
15   12.9      74    22.2
16   12.9      85    33.8
17   13.3      86    27.4
18   13.7      71    25.7
19   13.8      64    24.9
20   14.0      78    34.5
21   14.2      80    31.7
22   14.5      74    36.3
23   16.0      72    38.3
24   16.3      77    42.6
25   17.3      81    55.4
26   17.5      82    55.7
27   17.9      80    58.3
28   18.0      80    51.5
29   18.0      80    51.0
30   20.6      87    77.0

In [5]:
data.corr()

Girth   Height    Volume
Girth   1.000000  0.51928  0.967119
Height  0.519280  1.00000  0.598250
Volume  0.967119  0.59825  1.000000

In [6]:
data.shape

(31, 3)

In [ ]:
#Hypothesis testing
#H0: population correlation coeff is 0
#H1: Populatiojn correlation coeff is != 0

In [ ]:
#Parson product moment correlation hypothesis testing
#t statistic for testing t = correlation(x,y)*sqrt(N-2)/sqrt(1-correlation(x,y)^2)

In [5]:
t=0.967119*np.sqrt(31-2)/(np.sqrt(1-0.967119))

In [6]:
t

28.721452939679782

#p value is < 0.00001 for t value of 28.7 for 29 degress of freedom.


In [4]:
#So we reject the null hypothesis and say that population correlation coefficient is not 0.
#correlation between tree volume and tree girth is statistically siginificant